src/Microsoft.TestPlatform.Common/Hosting/TestRunTimeProviderManager.cs

```diff
@@ -73,7 +73,7 @@ public ITestRuntimeProvider GetTestHostManagerByUri(string hostUri)
             return null;
         }
 
-        public ITestRuntimeProvider GetTestHostManagerByRunConfiguration(RunConfiguration runConfiguration)
+        public ITestRuntimeProvider GetTestHostManagerByRunConfiguration(string runConfiguration)
```

Comment:
We want the implementer of ITestRuntimeProvider to have the full settings so they can decide on what ever parameters they want to, on whether or not can they Host test for this session.
This also allows them to add custom sections in runsettings, & process them here, rather than having the TP understand and process them.